text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,110 @@
+Rustdoc: stabilization of the `doc(cfg*)` attributes
+
+- Features Name: `doc_cfg`
+- Start Date: 2022-12-07
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#43781](https://github.com/rust-lang/rust/issues/43781)
+
+# Summary
+[summary]: #summary
+
+This RFC aims at providing to the rustdoc users the possibility to add visual markers in the rendered documentation to know under which conditions an item is available (currently possible through the following unstable features: `doc_cfg`, `doc_auto_cfg` and `doc_cfg_hide`).
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this RFC is to stabilize the possibility to add visual markers in the rendered documentation to know under which conditions an item is available.
+
+Providing this information to users will prevent a common issue: "why can I see this item in the documentation and yet can't use it in my code?".
+The end goal being to retrieve this information automatically so that the documentation maintenance cost won't increase.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC proposes to add the following attributes:
+
+ * `#[doc(cfg(...))]`
+ * `#[doc(auto_cfg)]`/`#[doc(no_auto_cfg)]`
+ * `#[doc(cfg_hide(...))]`
+
+### `#[doc(cfg(...))]`
+
+This attribute allows to manually add under which conditions an item is available to the documentation readers. Example:
+
+```rust
+// the "real" cfg condition
+#[cfg(feature = "futures-io")]
+// the `doc(cfg())` so it's displayed to the readers
+#[doc(cfg(feature = "futures-io"))]
+pub mod futures {}
+```
+
+It will display in the documentation for this module:
+
+![display of doc(cfg) feature](https://user-images.githubusercontent.com/81079/89731116-d7b7ce00-da44-11ea-87c6-022d192d6eca.png)
+
+This attribute works on modules and on items but cannot be used at the crate root level.
+
+### `#[doc(auto_cfg)]`/`#[doc(no_auto_cfg)]`
+
```

Comment:
This has the same problem as before, it's not very clear about how these attributes are applied.

Please at least add some examples showing them, and say somewhere that auto is the default.